components/concept/storage/src/main/java/mozilla/components/concept/storage/CreditCardsAddressesStorage.kt

```diff
@@ -106,12 +106,30 @@ interface CreditCardsAddressesStorage {
     suspend fun touchAddress(guid: String)
 }
 
+/**
+ * A credit card number. This structure exists to provider better typing at the API surface.
+ * @property number Either a plaintext or a ciphertext of the credit card number, depending on the type.
+ */
+sealed class CreditCardNumber(val number: String) {
+    /**
+     * An encrypted credit card number.
+     */
+    @Parcelize
+    data class Encrypted(private val data: String) : CreditCardNumber(data), Parcelable
+
+    /**
+     * A plaintext, non-encrypted credit card number.
+     */
+    data class Plaintext(private val data: String) : CreditCardNumber(data)
```

Comment:
Perhaps the CamelCase for PlainText?

```suggestion
    data class PlainText(private val data: String) : CreditCardNumber(data)
```

Reply:
In the context of cryptography, plaintext (as one word, vs "plain text") is an established and common term, so I think just `Plaintext` is fine.